perllib/FixMyStreet/App/Controller/Admin.pm

```diff
@@ -1856,6 +1858,43 @@ sub fetch_body_areas : Private {
     $c->stash->{fetched_areas_body_id} = $body->id;
 }
 
+sub update_extra_fields : Private {
+    my ($self, $c, $object) = @_;
+
+    my @indices = grep { /^metadata\[\d+\]\.code/ } keys %{ $c->req->params };
+    @indices = sort map { $_ =~ /(\d+)/ } @indices;
+
+    $object->set_extra_fields(());
+    foreach my $i (@indices) {
+        my $meta = {};
+        $meta->{code} = $c->get_param("metadata[$i].code");
+        next unless $meta->{code};
+        $meta->{order} = int $c->get_param("metadata[$i].order");
```

Comment:
Order isn't actually used anywhere after this. It looks like PopulateSeviceList sorts by the order upon import and then stores them in this order. So I guess this should do the same.